### ISSUE TITLE
Not working with tsserver?

### ISSUE THREAD

**Issue by jaivinwylde** [antosha417/nvim-lsp-file-operations#6]
After setting this up and trying to rename a TypeScript file, nothing happens and the debug also doesn't do anything when renaming. The plugin seems to be properly setup because there is an initial debug saying "Setting up nvim-tree integration"

Has tsserver support yet to be added?

**Comment by antosha417** [comment]
Hey, @jaivinwylde. I think `tsserver` does not listen for [workspace/WillRename](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_willRenameFiles) requests. 

I know that [typescript.nvim](https://github.com/jose-elias-alvarez/typescript.nvim) has command `:TypescriptRenameFile` that works with `tsserver`. So it should be possible to do. 

You are more than welcome to open a pr to support `tsserver`. Or I'll update this issue when I have a time to do it.

**Comment by antosha417** [comment]
The problem with `tsserver` is that when executing `_typescript.applyRenameFile` command instead of returning workspace edit as `metals` and `rust-analizer` do [it pushes workspace edit to the editor](https://github.com/typescript-language-server/typescript-language-server/blob/1daf209121fc20bbc0a64ec0491cd40582cb9a4b/src/lsp-server.ts#L1057).

And this edit is applied by neovim only after the files are moved by `nvim-tree`. But it should be applied before the move.

I don't see for now how to solve it on plugin side. Probably the easiest way would be to commit to tsserver support for `workspace/willRenameFiles` command. So that we can decide when to apply edits. The code for getting the edits already there. Just need to return them instead of applying them by seding `workspace/applyEdit` to the client.

**Comment by antosha417** [comment]
Hey, @jaivinwylde!
You can try it out if you install `typescript-language-server` from the pr branch. 

I'll wait for the next release of `typescript-language-server` and close the issue as completed.

**Comment by antosha417** [comment]
If you'll find any issues feel free to open another issue!
